src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,167 @@
+
+import { useContextMenu } from './hooks';
+
+import { ContextMenuRootProps } from './ContextMenu.props';
+import { ContextMenuMode } from './ContextMenu.enums';
+
+import { Trigger } from './components/Trigger/Trigger';
+import { Portal } from './components/Portal/Portal';
+import { Sub } from './components/Sub/Sub';
+import { SubTrigger } from './components/SubTrigger/SubTrigger';
+import { SubContent } from './components/SubContent/SubContent';
+import { Content } from './components/Content/Content';
+import { Arrow } from './components/Arrow/Arrow';
+import { Item } from './components/Item/Item';
+import { ItemRightSlot } from './components/ItemRightSlot/ItemRightSlot';
+import { Group } from './components/Group/Group';
+import { Label } from './components/Label/Label';
+import { CheckboxItem } from './components/CheckboxItem/CheckboxItem';
+import { ItemIndicator } from './components/ItemIndicator/ItemIndicator';
+import { RadioGroup } from './components/RadioGroup/RadioGroup';
+import { RadioItem } from './components/RadioItem/RadioItem';
+import { Separator } from './components/Separator/Separator';
+import { ItemIcon } from './components/ItemIcon/ItemIcon';
+import { SubRoot } from './components/SubRoot/SubRoot';
+import { FocusBlocker } from './components/FocusBlocker/FocusBlocker';
+
+import {
+  ContextMenuProvider,
+  ContextMenuRootProvider,
+  DISPLAY_NAME,
+} from './ContextMenu.context';
+
+import { ContextMenuHandle, ContextMenuType } from './ContextMenu.types';
+
+const HOVER_CLOSE_DELAY = 200;
+const ANIMATION_DURATION = 150;
+
+export const ContextMenu = forwardRef<ContextMenuHandle, ContextMenuRootProps>(
+  (
+    {
+      children,
+      mode,
+      onOpen,
+      isOpen,
+      defaultOpen,
+      onAnimatedOpen,
+      isCloseOnClick = true,
+      enableInnerInputFocus = false,
+      backgroundFocusBlockerContainers = [document.body],
+      backgroundFocusBlockerClassName,
+      backgroundInputFocusBlockerClassName,
+
+      ...rest
+    },
+    ref
+  ) => {
+    const {
+      mode: rootMode,
+      open,
+      onOpenChange,
+      triggerRef,
+      contentRef,
+      inheritedArrowColor,
+      animatedOpen,
+      animationDuration,
+      hoverCloseDelay,
+      temporaryHoverClose,
+      closeMenuImmediately,
+      onMouseEnter,
+      onMouseLeave,
+      enableTemporaryHoverClose,
+      onOpenByKeyboard,
+      onChildOpen,
+      onSubmenuOpen,
+      isRootContentBlocked,
+      isChildOpen,
+      itemWithFocusedInput,
+      setItemWithFocusedInput,
+    } = useContextMenu({
+      mode: mode as ContextMenuMode,
+      defaultOpen,
+      onOpen,
+      onAnimatedOpen,
+      animationDuration: ANIMATION_DURATION,
+      hoverCloseDelay: HOVER_CLOSE_DELAY,
+      isOpen,
+      enableInnerInputFocus,
+      backgroundFocusBlockerContainers,
+      backgroundFocusBlockerClassName,
+      backgroundInputFocusBlockerClassName,
+    });
+
+    useImperativeHandle(ref, () => ({
```

Comment:
I believe its a bad idea, ref should point to a dom node.

If you really need to give this methods to public api use it the way we do for button.

You can do the same way that we do Button.